core/src/main/java/org/elasticsearch/common/logging/DeprecationLogger.java

```diff
@@ -311,7 +311,7 @@ public static String formatWarning(final String s) {
      * @return the escaped string
      */
     public static String escape(String s) {
-        return s.replaceAll("(\\\\|\")", "\\\\$1");
+        return s.replaceAll("([\\\\\"])", "\\\\$1");
```

Comment:
Maybe we can get both worlds with `[\"\\\\]`; I think that this is as-readable as the current version, and we get the win of the character class change. Can you make this change and I can pull this in?